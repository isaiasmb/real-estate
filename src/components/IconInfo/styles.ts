import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: inline-flex;
    color: ${theme.colors.gray};
  `}
`

export const IconWrapper = styled.div`
  svg {
    width: 2rem;
  }
`

export const Value = styled.span`
  ${({ theme }) => css`
    margin-left: ${theme.spacings.xxxsmall};
  `}
`
