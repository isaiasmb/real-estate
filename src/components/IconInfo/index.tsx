import * as S from './styles'

export type IconInfo = {
  icon: JSX.Element
  value: string | number
}

const IconInfo = ({ icon, value }: IconInfo) => (
  <S.Wrapper>
    <S.IconWrapper>{icon}</S.IconWrapper>
    <S.Value>{value}</S.Value>
  </S.Wrapper>
)

export default IconInfo
