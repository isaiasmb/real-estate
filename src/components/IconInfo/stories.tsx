import { Story, Meta } from '@storybook/react/types-6-0'
import { Bed } from '@styled-icons/ionicons-outline/Bed'
import IconInfo from '.'

export default {
  title: 'IconInfo',
  component: IconInfo,
  argTypes: {
    icon: {
      type: ''
    }
  }
} as Meta

export const Default: Story = () => <IconInfo icon={<Bed />} value={5} />
