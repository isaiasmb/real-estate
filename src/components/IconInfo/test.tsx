import { screen } from '@testing-library/react'
import { Bed } from '@styled-icons/ionicons-outline/Bed'
import { renderWithTheme } from 'utils/tests/helpers'

import IconInfo from '.'

describe('<IconInfo />', () => {
  it('should render the icon and the value', () => {
    const { container } = renderWithTheme(
      <IconInfo icon={<Bed data-testid="icon" />} value={5} />
    )

    expect(screen.getByText('5')).toBeInTheDocument()
    expect(screen.getByTestId('icon')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
