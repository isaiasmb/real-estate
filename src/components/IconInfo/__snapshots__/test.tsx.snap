// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconInfo /> should render the icon and the value 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #8F8F8F;
}

.c1 svg {
  width: 2rem;
}

.c3 {
  margin-left: 0.4rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      data-testid="icon"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M384 240H96V136a40.12 40.12 0 0140-40h240a40.12 40.12 0 0140 40v104zM48 416V304a64.19 64.19 0 0164-64h288a64.19 64.19 0 0164 64v112"
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="32"
      />
      <path
        d="M48 416v-8a24.07 24.07 0 0124-24h368a24.07 24.07 0 0124 24v8M112 240v-16a32.09 32.09 0 0132-32h80a32.09 32.09 0 0132 32v16M256 240v-16a32.09 32.09 0 0132-32h80a32.09 32.09 0 0132 32v16"
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="32"
      />
    </svg>
  </div>
  <span
    class="c3"
  >
    5
  </span>
</div>
`;
